### PR TITLE
order and admin controllers were added

### PR DESCRIPTION
**Pull request by Victoria-Moiseienko**

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/carts/DeleteProductFromCartController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        Long id = Long.valueOf(req.getParameter("productId"));
+        Product product = productService.get(id);
+        shoppingCartService.deleteProduct(shoppingCartService.getByUserId(USER_ID),
+                product);
+
+        List<Product> productsInCart =
+                shoppingCartService.getByUserId(USER_ID).getProducts();
+
+        req.setAttribute("products", productsInCart);
+        req.getRequestDispatcher("/WEB-INF/views/carts/shoppingCart.jsp").forward(req, resp);
```
Comment (TimoxaV) [inline src/main/java/com/internet/shop/controllers/carts/DeleteProductFromCartController.java]: You are deleting product from cart and then getting all product from cart. Checklist says: "Single controller should be responsible for a single task". I guess you can just delete and then use GetCartProductsController to show products that are left in the cart.

Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/carts/DeleteProductFromCartController.java]: Updated. thank you

File: src/main/java/com/internet/shop/controllers/carts/DeleteProductFromCartController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long id = Long.valueOf(req.getParameter("productId"));
+        Product product = productService.get(id);
+        shoppingCartService.deleteProduct(shoppingCartService.getByUserId(USER_ID),
+                product);
+
+        resp.sendRedirect(req.getContextPath() + "/carts");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/carts/DeleteProductFromCartController.java]:
```suggestion
        resp.sendRedirect(req.getContextPath() + "/shopping-cart/info");
```

File: src/main/java/com/internet/shop/controllers/orders/CreateOrderController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            throws ServletException, IOException {
+        ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
+        if (shoppingCart.getProducts().isEmpty()) {
+            req.setAttribute("message", "Your cart is empty");
+            req.getRequestDispatcher("/WEB-INF/views/carts/shoppingCart.jsp")
+                    .forward(req,resp);
+        }
+
+        orderService.completeOrder(shoppingCart);
+        resp.sendRedirect(req.getContextPath() + "/orders/confirm");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/orders/CreateOrderController.java]: can we show `confirmOrder.jsp` here?

File: src/main/java/com/internet/shop/controllers/orders/DeleteOrderAdminController.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import com.internet.shop.model.Order;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteOrderAdminController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/orders/DeleteOrderAdminController.java]:
```suggestion
public class DeleteOrderController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/orders/DeleteOrderAdminController.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long id = Long.valueOf(req.getParameter("orderId"));
+
+        orderService.delete(id);
+        List<Order> orders = orderService.getAll();
+
+        req.setAttribute("orders", orders);
+        req.getRequestDispatcher("/WEB-INF/views/orders/admin.jsp").forward(req, resp);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/orders/DeleteOrderAdminController.java]: let's redirect to GetAllOrdersAdminController instead

File: src/main/java/com/internet/shop/controllers/orders/GetAllOrdersAdminController.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Order;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAllOrdersAdminController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/orders/GetAllOrdersAdminController.java]:
```suggestion
public class GetAllOrdersController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/products/AdminProductController.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AdminProductController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/products/AdminProductController.java]:
```suggestion
public class ManageProductsController extends HttpServlet {
```

File: src/main/webapp/WEB-INF/views/orders/userOrders.jsp [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        <th><a href="${pageContext.request.contextPath}/orders/admin">Orders Admin</a></th>
+        <th><a href="${pageContext.request.contextPath}/insertdata">Insert Data</a></th>
+    </tr>
+</table>
+<h1>user orders page</h1>
+<c:forEach var="order" items="${orders}">
+<table border="1">
+    <tr>
+        <td>
+            <c:out value="Order num  ${order.id} "/>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/orders/userOrders.jsp]:
```suggestion
            <c:out value="Order id  ${order.id} "/>
```

File: src/main/java/com/internet/shop/controllers/products/AdminProductController.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/products/AdminProductController.java]: after adding a new product to the db, we should redirect to this controller

File: src/main/webapp/web.xml [hunk 10]
```diff
@@ -15,70 +15,168 @@
...
-        <servlet-class>com.internet.shop.controllers.GetAllUsersController</servlet-class>
+        <servlet-name>insertData</servlet-name>
+        <servlet-class>com.internet.shop.controllers.InsertDataController</servlet-class>
     </servlet>
 
     <servlet-mapping>
-        <servlet-name>allUsers</servlet-name>
-        <url-pattern>/users/all</url-pattern>
+        <servlet-name>insertData</servlet-name>
+        <url-pattern>/insertdata</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/inject</url-pattern>
```

File: src/main/webapp/web.xml [hunk 11]
```diff
@@ -15,70 +15,168 @@
...
-        <servlet-class>com.internet.shop.controllers.RegistrationController</servlet-class>
+        <servlet-name>shoppingcart</servlet-name>
+        <servlet-class>com.internet.shop.controllers.carts.GetCartProductsController</servlet-class>
     </servlet>
 
     <servlet-mapping>
-        <servlet-name>registration</servlet-name>
-        <url-pattern>/users/registration</url-pattern>
+        <servlet-name>shoppingcart</servlet-name>
+        <url-pattern>/carts</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/shopping-cart/info</url-pattern>
```

File: src/main/webapp/web.xml [hunk 12]
```diff
@@ -15,70 +15,168 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addProductToCart</servlet-name>
+        <servlet-class>com.internet.shop.controllers.carts.AddProductToCartController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>addProductToCart</servlet-name>
+        <url-pattern>/carts/product/add</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/shopping-cart/products/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 13]
```diff
@@ -15,70 +15,168 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteProductFromCart</servlet-name>
+        <servlet-class>com.internet.shop.controllers.carts.DeleteProductFromCartController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>deleteProductFromCart</servlet-name>
+        <url-pattern>/carts/product/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/shopping-cart/products/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 14]
```diff
@@ -15,70 +15,168 @@
...
+        <url-pattern>/orders/details</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>getAllOrdersAdmin</servlet-name>
+        <servlet-class>com.internet.shop.controllers.orders.GetAllOrdersAdminController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>getAllOrdersAdmin</servlet-name>
+        <url-pattern>/orders/admin</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders</url-pattern>
```

File: src/main/webapp/web.xml [hunk 15]
```diff
@@ -15,70 +15,168 @@
...
+        <url-pattern>/orders/confirm</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>getUserOrders</servlet-name>
+        <servlet-class>com.internet.shop.controllers.orders.GetUserOrdersController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>getUserOrders</servlet-name>
+        <url-pattern>/orders/user</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/user/orders</url-pattern>
```

File: src/main/webapp/web.xml [hunk 16]
```diff
@@ -15,70 +15,168 @@
...
-        <servlet-class>com.internet.shop.controllers.ShowAllProductsController</servlet-class>
+        <servlet-name>allProduct</servlet-name>
+        <servlet-class>com.internet.shop.controllers.products.GetAllProductsController</servlet-class>
     </servlet>
 
     <servlet-mapping>
-        <servlet-name>showproduct</servlet-name>
-        <url-pattern>/products/showproducts</url-pattern>
+        <servlet-name>allProduct</servlet-name>
+        <url-pattern>/products/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products</url-pattern>
```

File: src/main/webapp/web.xml [hunk 17]
```diff
@@ -15,70 +15,168 @@
...
-        <servlet-class>com.internet.shop.controllers.GetCartProductsController</servlet-class>
+        <servlet-name>allProductAdmin</servlet-name>
+        <servlet-class>com.internet.shop.controllers.products.AdminProductController</servlet-class>
     </servlet>
 
     <servlet-mapping>
-        <servlet-name>shoppingcart</servlet-name>
-        <url-pattern>/carts</url-pattern>
+        <servlet-name>allProductAdmin</servlet-name>
+        <url-pattern>/products/admin/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/manage</url-pattern>
```

File: src/main/webapp/web.xml [hunk 18]
```diff
@@ -15,70 +15,168 @@
...
-        <servlet-class>com.internet.shop.controllers.AddProductToCartController</servlet-class>
+        <servlet-name>allUsers</servlet-name>
+        <servlet-class>com.internet.shop.controllers.users.GetAllUsersController</servlet-class>
     </servlet>
+
     <servlet-mapping>
-        <servlet-name>addProductToCart</servlet-name>
-        <url-pattern>/carts/addproduct</url-pattern>
+        <servlet-name>allUsers</servlet-name>
+        <url-pattern>/users/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/users</url-pattern>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controllers/carts/DeleteProductFromCartController.java (hunk 2 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/carts/DeleteProductFromCartController.java]: Done

File: src/main/java/com/internet/shop/controllers/orders/CreateOrderController.java (hunk 3 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/orders/CreateOrderController.java]: Done

File: src/main/java/com/internet/shop/controllers/orders/DeleteOrderAdminController.java (hunk 4 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/orders/DeleteOrderAdminController.java]: Done

File: src/main/java/com/internet/shop/controllers/orders/DeleteOrderAdminController.java (hunk 5 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/orders/DeleteOrderAdminController.java]: Done

File: src/main/java/com/internet/shop/controllers/orders/GetAllOrdersAdminController.java (hunk 6 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/orders/GetAllOrdersAdminController.java]: Done

File: src/main/java/com/internet/shop/controllers/products/AdminProductController.java (hunk 7 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/products/AdminProductController.java]: Done

File: src/main/webapp/WEB-INF/views/orders/userOrders.jsp (hunk 8 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/WEB-INF/views/orders/userOrders.jsp]: Done

File: src/main/webapp/web.xml (hunk 10 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml]: Done

File: src/main/java/com/internet/shop/controllers/products/AdminProductController.java (hunk 9 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/products/AdminProductController.java]: Done

File: src/main/webapp/web.xml (hunk 11 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 12 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 13 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 15 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 16 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 17 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 18 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml]: Done

File: src/main/java/com/internet/shop/controllers/orders/CreateOrderController.java [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
+        if (shoppingCart.getProducts().isEmpty()) {
+            req.setAttribute("message", "Your cart is empty");
+            req.getRequestDispatcher("/WEB-INF/views/carts/shoppingCart.jsp")
+                    .forward(req,resp);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/orders/CreateOrderController.java:28]:
```suggestion
            req.getRequestDispatcher("/WEB-INF/views/carts/shoppingCart.jsp")
                    .forward(req,resp);
                    return;
```

File: src/main/webapp/WEB-INF/views/carts/shoppingCart.jsp [hunk 20]
```diff
@@ -8,13 +8,18 @@
...
-        <th><a href="${pageContext.request.contextPath}/products/showproducts">Product list</a></th>
-        <th><a href="${pageContext.request.contextPath}/carts">Shopping cart</a></th>
+        <th><a href="${pageContext.request.contextPath}/products">Product list</a></th>
+        <th><a href="${pageContext.request.contextPath}/products/manage">Product list Admin</a></th>
+        <th><a href="${pageContext.request.contextPath}/shopping-cart/info">Shopping cart</a></th>
+        <th><a href="${pageContext.request.contextPath}/user/orders">Orders</a></th>
+        <th><a href="${pageContext.request.contextPath}/orders">Orders Admin</a></th>
+        <th><a href="${pageContext.request.contextPath}/inject">Inject</a></th>
     </tr>
 </table>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/carts/shoppingCart.jsp:20]: you can use `<%@include file="header.jsp" %>`

File: src/main/webapp/WEB-INF/views/orders/admin.jsp [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+        <th><a href="${pageContext.request.contextPath}/orders">Orders Admin</a></th>
+        <th><a href="${pageContext.request.contextPath}/inject">Inject</a></th>
+    </tr>
+</table>
+<h1>Orders for admin</h1>
+<c:forEach var="order" items="${orders}">
+    <table border="1">
+        <tr>
+            <td>
+                <c:out value="Order num  ${order.id}"/>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/orders/admin.jsp:26]:
```suggestion
                <c:out value="Order id  ${order.id}"/>
```

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**